artifactory/utils/dotnet/toolchaincmd.go

```diff
@@ -0,0 +1,102 @@
+package dotnet
+
+import (
+	"github.com/jfrog/jfrog-cli/utils/cliutils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"io"
+	"os/exec"
+)
+
+type ToolchainType int
+
+const (
+	Nuget ToolchainType = iota
+	Dotnet
+)
+
+var CmdTypes = []string{
+	"nuget",
+	"dotnet",
+}
+
+func (toolchainType ToolchainType) String() string {
+	return CmdTypes[toolchainType]
+}
+
+var CmdFlagPrefixes = []string{
+	"-",
+	"--",
+}
+
+func (toolchainType ToolchainType) GetTypeFlagPrefix() string {
+	return CmdFlagPrefixes[toolchainType]
+}
+
+var AddSourceArgs = [][]string{
+	{"sources", "add"},
+	{"nuget", "add", "source"},
+}
```

Comment:
We are maintaining here several arrays, with two elements, one of each toolchain. All of these arrays are loosely bound to the ToolchainType consts. If the elements order in any of these arrays is flipped, a bug will be created.
I suggest taking a more object oriented approach, and use a Toolchain struct, whch will include a field for each of the tool chain characteristics. We'll need to two Toolchain structs - one for Nuget and one for DotnetCore.